Engine/ScriptAnalyzer.cs

```diff
@@ -24,14 +24,25 @@
 using System.Management.Automation.Runspaces;
 using System.Reflection;
 using System.Globalization;
+using System.Collections.Concurrent;
+using System.Threading.Tasks;
 
 namespace Microsoft.Windows.PowerShell.ScriptAnalyzer
 {
-    internal class ScriptAnalyzer
+    public sealed class ScriptAnalyzer
     {
-        #region Private memebers
+        #region Private members
 
+        private IOutputWriter outputWriter;
         private CompositionContainer container;
+        Dictionary<string, List<string>> validationResults = new Dictionary<string, List<string>>();
+        Dictionary<string, List<RuleSuppression>> ruleSuppressions;
```

Comment:
Seems like this is only used in AnalyzeSyntaxTree. We should remove the declaration here and move it there.